File: Code/Visitor.dotNet/Visitor.dotNet/Visitor.dotNet/Visitors/PrintVisitor.cs

using Visitor.Models;

namespace Visitor.Visitors
{
    internal class PrintVisitor : IVisitor
    {
        public void Visit(VisitableSquare s)
        {
            Console.WriteLine($"This square has side {s.Side}");
        }

        public void Visit(Rectangle r)
        {
            Console.WriteLine($"This rectangle has width {r.Width} amd height {r.Height}");
        }
    }
}

File: Code/Visitor.Java/src/visitors/PrintVisitor.java
package visitors;

import models.*;
public class PrintVisitor implements IVisitor {

	@Override
	public void visit(VisitableSquare s) {
		System.out.println("This square has side " + s.getSide());
	}

	@Override
	public void visit(Rectangle r) {
		System.out.println("This rectangle has width " + r.getWidth() + " and height " + r.getHeight());
	}

}
